apps/common/lib/lexical/ast/analysis/scope.ex

```diff
@@ -23,12 +25,22 @@ defmodule Lexical.Ast.Analysis.Scope do
         }
 
   def new(%__MODULE__{} = parent_scope, id, %Range{} = range, module \\ []) do
+    uses =
+      if module == parent_scope.module do
+        # if we're still in the same module, we have the same uses
```

Comment:
Even if we're in an inner module, should we consider that we have the same uses? We do that for aliases, imports, requires. 

```elixir
defmodule Parent do
  use SomeModule

  defmodule Child do
    # anything imported/required/etc. from `use SomeModule` is still available here
  end
end
```

Reply:
this is the trick, uses can do a lot more than import and require, they can add functions to the current module and greatly affect its behavior, this is not carried through to nested modules. 
If we want to actually be accurate for imports and requires, we need to introspect the module.